keycloak-controller/src/main/java/io/enmasse/keycloak/controller/KeycloakParams.java

```diff
@@ -15,28 +15,61 @@
  */
 package io.enmasse.keycloak.controller;
 
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.openssl.PEMParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileReader;
+import java.io.StringReader;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
 import java.util.Map;
 
 public class KeycloakParams {
 
+    private static final Logger log = LoggerFactory.getLogger(KeycloakParams.class);
+
     private final String host;
     private final int httpPort;
     private final String adminUser;
     private final String adminPassword;
+    private final KeyStore keyStore;
 
-    public KeycloakParams(String host, int httpPort, String adminUser, String adminPassword) {
+    public KeycloakParams(String host, int httpPort, String adminUser, String adminPassword, KeyStore keyStore) {
         this.host = host;
         this.httpPort = httpPort;
         this.adminUser = adminUser;
         this.adminPassword = adminPassword;
+        this.keyStore = keyStore;
     }
 
-    public static KeycloakParams fromEnv(Map<String, String> env) {
+    public static KeycloakParams fromEnv(Map<String, String> env) throws Exception {
         String host = getEnvOrThrow(env, "STANDARD_AUTHSERVICE_SERVICE_HOST");
-        int httpPort = Integer.parseInt(getEnvOrThrow(env, "STANDARD_AUTHSERVICE_SERVICE_PORT_HTTP"));
+        int httpPort = Integer.parseInt(getEnvOrThrow(env, "STANDARD_AUTHSERVICE_SERVICE_PORT_HTTPS"));
         String adminUser = getEnvOrThrow(env, "STANDARD_AUTHSERVICE_ADMIN_USER");
         String adminPassword = getEnvOrThrow(env, "STANDARD_AUTHSERVICE_ADMIN_PASSWORD");
-        return new KeycloakParams(host, httpPort, adminUser, adminPassword);
+        KeyStore keyStore = createKeyStore(env);
+
+        return new KeycloakParams(host, httpPort, adminUser, adminPassword, keyStore);
+    }
+
+    private static KeyStore createKeyStore(Map<String, String> env) throws Exception {
+        String authServiceCa = getEnvOrThrow(env, "STANDARD_AUTHSERVICE_CA_CERT");
+        try (PEMParser pemParser = new PEMParser(new StringReader(authServiceCa))) {
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(null);
+            JcaX509CertificateConverter converter = new JcaX509CertificateConverter();
+            Certificate certificate = converter.getCertificate((X509CertificateHolder) pemParser.readObject());
+            keyStore.setCertificateEntry("standard-authservice", certificate);
+
+            return keyStore;
+        } catch (Exception ignored) {
+            log.warn("Error creating keystore for authservice CA", ignored);
+            throw ignored;
+        }
     }
```

Comment:
Do we actually need to use bouncycastle here, for instance https://github.com/apache/qpid-broker-j/blob/master/broker-core/src/main/java/org/apache/qpid/server/transport/network/security/ssl/SSLUtil.java#L310-L330 reads X509 certificates from PEM files without needing bouncycastle

Reply:
Didn't know about that, thanks!